wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -285,10 +285,11 @@ def _etag_from_obj(
         ]
     ) -> ETag:
         etag: ETag
-        if hasattr(obj, "e_tag"):
+        boto = util.get_module("boto3")
```

Comment:
If we don't import the `boto3` module here then there's no way to catch the exception, other than using the catchall `except:`

Reply:
you should just store the boto3 reference in `init_boto` rather than reimporting it here

Reply:
shouldn't need this anymore.